src/TestFramework/TestFramework/Assertions/CollectionAssert.cs

```diff
@@ -418,9 +418,25 @@ public static void IsSubsetOf([NotNull] ICollection? subset, [NotNull] ICollecti
     {
         Assert.CheckParameterNotNull(subset, "CollectionAssert.IsSubsetOf", "subset", string.Empty);
         Assert.CheckParameterNotNull(superset, "CollectionAssert.IsSubsetOf", "superset", string.Empty);
-        if (!IsSubsetOfHelper(subset, superset))
+        Tuple<bool, ICollection<object?>> isSubsetValue = IsSubsetOfHelper(subset, superset);
+        if (!isSubsetValue.Item1)
         {
-            Assert.ThrowAssertFailed("CollectionAssert.IsSubsetOf", Assert.BuildUserMessage(message, parameters));
+            var sb = new StringBuilder();
+            foreach (object? item in isSubsetValue.Item2)
+            {
+                sb.Append(' ').Append(Convert.ToString(item, CultureInfo.InvariantCulture)).Append(',');
+            }
+
+            string returnedSubsetValueMessage = sb.ToString();
+
+            // Remove the last trailing comma
+            if (!string.IsNullOrEmpty(returnedSubsetValueMessage))
+            {
+                returnedSubsetValueMessage = returnedSubsetValueMessage.TrimEnd(',').Trim();
+            }
```

Comment:
Are we re-implementing String.Join here (or pre-filtered collection, and then string.join)?